src/indexing.rs

```diff
@@ -39,57 +43,59 @@ impl SpIndex for usize {
         self
     }
 
+    #[inline(always)]
+    fn index_unchecked(self) -> usize {
+        self
+    }
+
     #[inline(always)]
     fn from_usize(ind: usize) -> Self {
         ind
     }
+
+    #[inline(always)]
+    fn from_usize_unchecked(ind: usize) -> Self {
+        ind
+    }
 }
 
-macro_rules! sp_index_signed_impl {
+macro_rules! sp_index_impl {
     ($int: ident) => {
         impl SpIndex for $int {
             #[inline(always)]
             fn index(self) -> usize {
-                debug_assert!(self >= 0);
-                self as usize
+                num_traits::cast(self).unwrap_or_else(|| {
+                    panic!("Failed to convert {} to usize", self)
+                })
             }
 
             #[inline(always)]
-            fn from_usize(ind: usize) -> Self {
-                let max = $int::max_value() as usize;
-                debug_assert!(ind <= max);
-                ind as $int
+            fn index_unchecked(self) -> usize {
+                self as usize
             }
-        }
-    };
-}
-
-sp_index_signed_impl!(isize);
-sp_index_signed_impl!(i64);
-sp_index_signed_impl!(i32);
-sp_index_signed_impl!(i16);
 
-macro_rules! sp_index_unsigned_impl {
-    ($int: ident) => {
-        impl SpIndex for $int {
             #[inline(always)]
-            fn index(self) -> usize {
-                self as usize
```

Comment:
btw since this macro is called on `u64`, it seems to assume `usize` to be at least 64 bits, which is not true for 32 bit builds?

Reply:
Yes I've overlooked this issue when implementing the `SpIndex` trait it seems. Looks like with your new overflow check this issue should be gone.